backend/app/api/v1/endpoints/ols/olsJobs.py

```diff
@@ -42,10 +42,16 @@ async def jobs(
     sort: str = Query(None, description="To sort fields on specified direction"),
     filter: str = Query(None, description="Query to filter the jobs"),
 ):
-
-    return await ocpJobs(start_date, end_date, 
-                         pretty, size, offset, sort, 
-                         filter="benchmark='ols-load-generator'")
+    filter_clause = "benchmark='ols-load-generator'"
+    filter = (
+        filter_clause
+        if not filter
+        else filter
+        if filter_clause in filter
```

Comment:
I briefly considered worrying about that case and decided not to; but now you're making me reconsider. On the other hand, while I believe the standard UI action will indeed use exactly `benchmark='ols-load-generator'`, I don't think that `TableFilter` will actually manifest a filter option if `ols-load-generator` is the only `benchmark` value. And the filter parser would generate the same match JSON with `benchmark=ols-load-generator` or `benchmark="ols-load-generator"` or `benchmark=['ols-load-generator']` and I really didn't want to deal with that.

So this attempted query optimization probably doesn't hurt, but might not really help much, either... and my way kept it all on one line. 😁

Reply:
yeah, just thought if I am doing it let's get it done completely.